test/FunctionsV2/OutOfProcTests.cs

```diff
@@ -122,5 +138,209 @@ public async Task CallHttpActionOrchestrationWithManagedIdentity()
+                    Assert.True(rpcBaseUri.IsLoopback);
+                    Assert.Equal("http", rpcBaseUri.Scheme);
+                    Assert.Equal("/durabletask/", rpcBaseUri.AbsolutePath);
+                }
+                else
+                {
+                    Assert.Equal(JTokenType.Null, rpcBaseUrl.Type);
+                }
+            }
+        }
+
+        [Theory]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        [InlineData(null, true)]
+        [InlineData("node", true)]
+        [InlineData("java", true)]
+        [InlineData("powershell", true)]
+        [InlineData("python", true)]
+        [InlineData("dotnet", false)]
+        public async Task TestLocalRcpEndpointRuntimeVersion(string runtimeVersion, bool enabledExpected)
+        {
+            INameResolver nameResolver = new SimpleNameResolver(
+                new Dictionary<string, string>
+                {
+                    { "FUNCTIONS_WORKER_RUNTIME", runtimeVersion },
+                });
+
+            using (ITestHost host = TestHelpers.GetJobHost(
+                this.loggerProvider,
+                nameof(this.TestLocalRcpEndpointRuntimeVersion),
+                enableExtendedSessions: false,
+                localRpcEndpointEnabled: null /* use FUNCTIONS_WORKER_RUNTIME to decide */,
+                nameResolver: nameResolver))
+            {
+                await host.StartAsync();
+
+                // Check to see whether the local RPC endpoint has been opened
+                IPGlobalProperties ipGlobalProperties = IPGlobalProperties.GetIPGlobalProperties();
+                IPEndPoint[] endpoints = ipGlobalProperties.GetActiveTcpListeners();
+
+                const string LocalRcpAddress = "127.0.0.1:17071";
+                if (enabledExpected)
+                {
+                    Assert.Contains(LocalRcpAddress, endpoints.Select(ep => ep.ToString()));
+                }
+                else
+                {
+                    Assert.DoesNotContain(LocalRcpAddress, endpoints.Select(ep => ep.ToString()));
+                }
+
+                await host.StopAsync();
+            }
+        }
+
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task InvokeLocalRpcEndpoint()
+        {
+            using (ITestHost host = TestHelpers.GetJobHost(
+                this.loggerProvider,
+                nameof(this.InvokeLocalRpcEndpoint),
+                enableExtendedSessions: false,
+                localRpcEndpointEnabled: true,
+                notificationUrl: null))
+            {
+                await host.StartAsync();
+
+                using (var client = new WebClient())
+                {
+                    string jsonString = client.DownloadString("http://localhost:17071/durabletask/instances");
+
+                    // The data doesn't matter, but the result should be an array
```

Comment:
This comment is just meant to indicate this test does not care about the data because this test is just making sure the request is actually going through, right? Presumably the schema of this data is expected to match the webhook call.

Reply:
Yes, this is why I'm doing `JArray.Parse(jsonString)`. I'm expecting the result to be an array and this will throw if it's not an array. In theory it should be an empty array, but whether it is empty or not doesn't really matter from the POV of this test.